src/helpers.rs

```diff
@@ -348,4 +349,42 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
         Ok(())
     }
+
+    fn read_os_string(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx, OsString> {
+        let bytes = self.eval_context_mut().memory().read_c_str(scalar)?.to_vec();
+        if cfg!(unix) {
+            Ok(std::os::unix::ffi::OsStringExt::from_vec(bytes))
+        } else {
+            std::str::from_utf8(&bytes)
+                .map_err(|_| err_unsup_format!("{:?} is not a valid utf-8 string", bytes).into())
+                .map(OsString::from)
+        }
+    }
+
+    fn write_os_string(&mut self, os_string: OsString, ptr: Pointer<Tag>, size: u64) -> InterpResult<'tcx> {
```

Comment:
This should take an `OsStr` as argument (`OsString` and `OsStr` are like `String` and `str`).

Reply:
Oh and the concern with the name applies here to. This is used to emulate Unix APIs that *return* a string, and we have to write things in whatever way they expect. This would fail horribly if it was used in a Windows API emulation.

The format on Unix is null-terminated sequences of bytes, and so that's what the function works with.

Reply:
I have two candidates: `read_unix_string` or `read_os_string_from_unix_bytes`.

Reply:
I think the last commit solves the `OsString` part of your comments. We can keep the name discussion in the other conversation.